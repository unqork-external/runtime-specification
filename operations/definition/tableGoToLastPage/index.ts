export * from './tableGoToLastPage.model'
export * from './tableGoToLastPage.options'
