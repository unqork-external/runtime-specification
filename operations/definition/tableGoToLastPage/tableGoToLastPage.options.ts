import { Description, Required } from '@tsed/schema'

export class TableGoToLastPageOperationOptions {
  @Required()
  @Description('Key of the Table component to change pagination position of')
  targetKey: string
}
