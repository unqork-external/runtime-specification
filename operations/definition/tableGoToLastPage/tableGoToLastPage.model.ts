import { DiscriminatorValue, Required } from '@tsed/schema'

import { TableGoToLastPageOperationOptions } from './tableGoToLastPage.options'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.TABLE_GO_TO_LAST_PAGE)
export class TableGoToLastPageOperation extends Operation<OperationTypes.TABLE_GO_TO_LAST_PAGE> {
  @Required()
  public options: TableGoToLastPageOperationOptions = new TableGoToLastPageOperationOptions()
}
