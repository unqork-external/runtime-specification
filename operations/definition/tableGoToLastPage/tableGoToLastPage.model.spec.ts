import { TableGoToLastPageOperation } from './tableGoToLastPage.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: TableGoToLastPage Tests', () => {
  let validate, schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(TableGoToLastPageOperation)
    validate = response.validate
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
