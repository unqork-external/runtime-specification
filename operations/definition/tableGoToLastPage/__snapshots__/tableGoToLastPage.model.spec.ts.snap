// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: TableGoToLastPage Tests should match this JSON Schema 1`] = `
{
  "definitions": {
    "TableGoToLastPageOperationOptions": {
      "properties": {
        "targetKey": {
          "description": "Key of the Table component to change pagination position of",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
      ],
      "type": "object",
    },
  },
  "description": "Navigates to the last page of a paginated Table component.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/TableGoToLastPageOperationOptions",
    },
    "type": {
      "const": "TABLE_GO_TO_LAST_PAGE",
      "examples": [
        "TABLE_GO_TO_LAST_PAGE",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "stability": "STABLE",
  "type": "object",
}
`;
